pkg/library/libraryapplyconfiguration/equivalence.go

```diff
@@ -84,52 +87,117 @@ func EquivalentClusterApplyResultIgnoringEvents(field string, lhs, rhs SingleClu
 	return qualifiedReasons
 }
 
-func reasonForDiff(nameOfDestination string, missingInDestination []manifestclient.SerializedRequest, allSourceRequests, allDestinationRequests []manifestclient.SerializedRequestish) []string {
+// expandedMetadata is useful for describing diffs, potentially to get pushed into manifestclient
+type expandedMetadata struct {
+	metadata               manifestclient.ActionMetadata
+	fieldManager           string
+	controllerInstanceName string
+}
+
+func expandedMetadataFor(serializedRequest *manifestclient.SerializedRequest) expandedMetadata {
+	if serializedRequest == nil {
+		return expandedMetadata{}
+	}
+	metadata := serializedRequest.GetLookupMetadata()
+	fieldManager := ""
+	controllerInstanceName := ""
+
+	isApply := serializedRequest.Action == manifestclient.ActionApply || serializedRequest.Action == manifestclient.ActionApplyStatus
+	if isApply {
+		lhsOptions := &metav1.ApplyOptions{}
+		if err := yaml.Unmarshal(serializedRequest.Options, lhsOptions); err == nil {
+			// ignore err.  if it doesn't work we get the zero value and that's ok
```

Comment:
not being able to deserialise the options sounds like a serious issue. do we really want to ignore it ?

Reply:
> not being able to deserialise the options sounds like a serious issue. do we really want to ignore it ?

Ignoring here means we will proceed with the diff, not that we skip it.